### PR TITLE
Fix some constructor-related generation refactorings with top-level nullability

### PR DESCRIPTION
**Pull request by jasonmalinowski**
Commit at a time may be useful here.

### CODE REVIEW

File: src/Workspaces/Core/Portable/Shared/Extensions/SyntaxGeneratorExtensions.cs [hunk 1]
```diff
@@ -125,7 +125,7 @@ internal static partial class SyntaxGeneratorExtensions
             foreach (var parameter in parameters)
             {
                 var refKind = parameter.RefKind;
-                var parameterType = parameter.Type;
+                var parameterType = parameter.Type.WithNullability(parameter.NullableAnnotation);
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Shared/Extensions/SyntaxGeneratorExtensions.cs]: :'(

File: src/Features/Core/Portable/InitializeParameter/AbstractInitializeMemberFromParameterCodeRefactoringProviderMemberCreation.cs [hunk 2]
```diff
@@ -133,7 +133,7 @@ internal abstract partial class AbstractInitializeMemberFromParameterCodeRefacto
                         default,
                         accessibilityLevel,
                         DeclarationModifiers.ReadOnly,
-                        parameter.Type, uniqueName);
+                        parameter.Type.WithNullability(parameter.NullableAnnotation), uniqueName);
```
Comment (ryzngard) [inline src/Features/Core/Portable/InitializeParameter/AbstractInitializeMemberFromParameterCodeRefactoringProviderMemberCreation.cs]:
```suggestion
                        parameter.GetTypeWithAnnotatedNullability(), uniqueName);
```

File: src/Features/Core/Portable/InitializeParameter/AbstractInitializeMemberFromParameterCodeRefactoringProviderMemberCreation.cs [hunk 3]
```diff
@@ -186,7 +186,7 @@ private static string GenerateUniqueName(IParameterSymbol parameter, ImmutableAr
                         default,
                         accessibilityLevel,
                         new DeclarationModifiers(),
-                        parameter.Type,
+                        parameter.Type.WithNullability(parameter.NullableAnnotation),
```
Comment (ryzngard) [inline src/Features/Core/Portable/InitializeParameter/AbstractInitializeMemberFromParameterCodeRefactoringProviderMemberCreation.cs]:
```suggestion
                        parameter.GetTypeWithAnnotatedNullability(),
```

File: src/Workspaces/Core/Portable/Shared/Extensions/SyntaxGeneratorExtensions.cs (hunk 1 above)
Comment (ryzngard) [inline src/Workspaces/Core/Portable/Shared/Extensions/SyntaxGeneratorExtensions.cs]:
```suggestion
                var parameterType = parameter.GetTypeWithAnnotatedNullability();
```

File: src/Features/Core/Portable/InitializeParameter/AbstractInitializeMemberFromParameterCodeRefactoringProviderMemberCreation.cs (hunk 2 above)
Comment (jasonmalinowski) [reply src/Features/Core/Portable/InitializeParameter/AbstractInitializeMemberFromParameterCodeRefactoringProviderMemberCreation.cs]: Oh right, that's what @CyrusNajmabadi was complaining about. Fixed.

**Comment by CyrusNajmabadi** [comment]
You couldn't tell from my tears? :D

**Comment by jasonmalinowski** [comment]
@CyrusNajmabadi It was hard to know precisely which thing you were crying about!

**Comment by CyrusNajmabadi** [comment]
Mostly about eating too much cheese this weekend :D

**Comment by CyrusNajmabadi** [comment]
🧀 🥂

**Pull request merged**